core/commands/UserResetPasswordCommand.php

```diff
@@ -49,30 +55,41 @@ function __construct( array $p_data ) {
 
 	/**
 	 * Validate the data.
+	 * @throws ClientException
 	 */
 	function validate() {
-		$this->user_id_reset = (int)$this->query( 'id', null );
-		if( $this->user_id_reset <= 0 || !user_exists( $this->user_id_reset ) ) {
-			throw new ClientException( 'Invalid user id', ERROR_INVALID_FIELD_VALUE, array( 'id' ) );
-		}
-
-		# Ensure user has access level to delete users
+		# Ensure user has the required access level to reset passwords
 		if( !access_has_global_level( config_get_global( 'manage_user_threshold' ) ) ) {
 			throw new ClientException( 'Access denied to reset user password', ERROR_ACCESS_DENIED );
 		}
 
+		$this->user_id_reset = (int)$this->query( 'id', null );
+
+		# Make sure the account exists
 		$t_user = user_get_row( $this->user_id_reset );
-		if( $t_user === false ) { // cannot be
+		if( $t_user === false ) {
 			throw new ClientException( 'Invalid user id', ERROR_INVALID_FIELD_VALUE, array( 'id' ) );
 		}
 
-		# Ensure that the account to be reset is of equal or lower access to the
-		# current user.
+		# Mantis can't reset protected accounts' passwords, but if the
+		# account is locked, we allow the operation as Unlock
+		if( auth_can_set_password( $this->user_id_reset )
```

Comment:
Why are we ANDing these checks?  I would expect an error if `auth_can_set_password()` is false.  I also expect a separate check that errors out if account is protected.  I don't think we should worry about user can sign in or not (i.e. remove user_is_login_request_allowed).

Reply:
This is just replicating the logic as it was before the Command was introduced. Without this, there would be no way to unlocked a protected account. 

The user_is_login_request_allowed() calls basically checks whether the account is locked or not. This will disappear when the new User Unlock command is added.